astropy/table/table.py

```diff
@@ -1193,6 +1246,23 @@ def add_columns(self, cols, indexes=None, copy=True):
                 new_indexes.insert(i, None)
                 newcols.insert(i, col)
 
+        if rename_duplicate:
+            colnames = set(col_getattr(col, 'name') for col in newcols)
+            if len(colnames) != len(newcols):
+                final_names = set(self.colnames)
+                for col in cols:
+                    name = col_getattr(col, 'name')
+                    if name in final_names:
+                        i = 1
+                        newname = '{0}_{1}'.format(name, i)
+                        while newname in final_names:
+                            i += 1
+                            newname = '{0}_{1}'.format(name, i)
+                        col_setattr(col, 'name', newname)
+                        warnings.warn("Column '{0}' is renamed to '{1}'"
+                                      .format(name, newname), AstropyUserWarning)
+                        final_names.add(newname)
+
```

Comment:
I guess it's never a bottleneck, but is it necessary to set the name unnecessary n-1 times (w/ ``col_setattr()``)?